02.calendar/cal.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'date'
+require 'optparse'
+
+CALENDER_WIDTH = 20
+DAY_DIGIT = 2
+MIN_YEAR = 1
+MAX_YEAR = 9999
+
+def main
+  today = Date.today
+  opt = OptionParser.new
+  opt.on('-m')
+  opt.on('-y')
+  params = ARGV.getopts('', "y:#{today.year}", "m:#{today.month}")
+  year = params['y'].to_i
+  month = params['m'].to_i
+
+  unless valid_year?(year)
+    puts "cal: year `#{year}' not in range #{MIN_YEAR}..#{MAX_YEAR}"
+    return
+  end
+
+  unless valid_month?(month)
+    puts "cal: #{month} is neither a month number (1..12) nor a name"
+    return
+  end
+
+  calender_header(year, month)
+  dates = generate_dates(year, month)
+  calender_body(today, dates)
+end
+
+def valid_year?(year)
+  year >= MIN_YEAR && year <= MAX_YEAR
+end
+
+def valid_month?(month)
+  month >= 1 && month <= 12
+end
+
+def generate_dates(year, month)
+  first_date = Date.new(year, month, 1)
+  last_date = Date.new(year, month, -1)
+  (first_date..last_date).to_a
+end
+
+def calender_header(year, month)
+  puts "#{month}月 #{year}".center(CALENDER_WIDTH)
+  puts '日 月 火 水 木 金 土'
+end
+
+def formatted_day(date, today)
```

Comment:
```suggestion
def format_day(date, today)
```

の方が自然かなと思います。（過去分詞形だと「整形された日」のように形容詞のように見えてしまう）